### PR TITLE
MHV-53672 Ensure-SideNav-Highlights-For-Details-Pages

### PR DESCRIPTION
**Pull request by tynyttie**
## Summary
Updated SideNav to highlight when on detail pages.

## Related issue(s)
[MHV-53672](https://jira.devops.va.gov/browse/MHV-53672) - Ensure SideNav highlights for details pages

Ensure SideNav highlights for details pages

When a user browses to a details page (e.g. Allergies Details), the appropriate link should be highlighted in the sidenav. This must work for all domains.

This should be addressable generically. It looks like we have a function to do this in Navigtion.jsx called handleActiveLinksStyle. Make sure it is functioning correctly.

Picture from Lexi of it not working:

GIVEN: A user is on the list page

WHEN: A user navigates to the details page

THEN: The appropriate side navigation is highlighted

 

Feature Flag Y/N ? N

Manual Testing Y/N ? Y-Sanjay

Automated Testing Y/N ? Y-Timothy

Accessibility Testing Y/N ? Y If necessary

UCD Validation Y/N Y

Acceptance Criteria:

AC1 The appropriate link should be highlighted in the side nav when navigating to details page

AC2 All domains have been updated

AC3 UCD validation is required

AC4

Links to UCD:

Sketch Link:

Other Design Notes

 

Architectural Notes:

 

Test Coverage
No Tests were found testing the requirement.

## Testing done
Conducted UI testing to ensure the expected behavior. I updated existing unit test for new components.

## Screenshots
Screenshot 2024-02-21 at 12 24 39 PM (2)
Screenshot 2024-02-21 at 12 24 43 PM (2)

## What areas of the site does it impact?

All detail pages 

## Acceptance criteria
Ensure SideNav highlights for details pages

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/developer-docs/wcag-2-1-success-criteria-and-foundational-testing) has been performed

### Authentication

- [x] Did you login to a local build and verify all authenticated routes work as expected with a test user

### CODE REVIEW

File: src/applications/mhv/medical-records/containers/App.jsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable no-console */
```
[the remark this answers is not in the document]
Comment (tynyttie) [reply src/applications/mhv/medical-records/containers/App.jsx]: fixed

File: src/applications/mhv/medical-records/containers/App.jsx [hunk 2]
```diff
@@ -105,26 +105,50 @@ const App = ({ children }) => {
...
+
+        const activePath = currentPath.startsWith(allergiesPath)
+          ? currentPath
+          : allergiesPath;
+        navPaths[0].subpaths.push({
+          path: activePath,
+          label: 'Allergies and reactions',
+          datatestid: 'allergies-sidebar',
+        });
+      }
```
Comment (mmoyer-va) [inline src/applications/mhv/medical-records/containers/App.jsx:148]: When in the details page, this code updates the link to point to the current record. The links in the sidenav should always point to the list page.
image

**Review by mmoyer-va — CHANGES_REQUESTED** [review]
Really this functionality should be handled by the `handleActiveLinksStyle` function inside `Navigation`. The links as created in `App` should be static and always point to the list view.

It should also be generic. Because all of our list/details URLs have the same structure for every domain, we should be able to have one bit of code handle every scenario.

File: src/applications/mhv/medical-records/containers/App.jsx (hunk 2 above)
Comment (tynyttie) [reply src/applications/mhv/medical-records/containers/App.jsx]: Fixed it.

**Review by mmoyer-va — CHANGES_REQUESTED** [review]
Can you write some tests for this function? I think the easiest way would be to split this into two functions:
```
const getActiveLinksStyle = (linkPath, currentPath) => {
  ...all the code
}

const handleActiveLinksStyle = path => {
  return getActiveLinksStyle (path, location.pathname);
}
```
Then you can write some easy tests for `handleActiveLinksStyle`. I would write tests for all of our routes. Include some negative tests as well.

**Review by mmoyer-va — DISMISSED** [review]
Looks good!

**Review by mmoyer-va — DISMISSED** [review]

**Review by micahchiang — DISMISSED** [review]

**Review by mmoyer-va — DISMISSED** [review]

**Review by timothy-steele-va — DISMISSED** [review]

**Review by mmoyer-va — DISMISSED** [review]

**Review by mmoyer-va — APPROVED** [review]

**Pull request merged**